dashboard/app/jobs/aichat_request_chat_completion_job.rb

```diff
@@ -18,7 +18,12 @@ class AichatRequestChatCompletionJob < ApplicationJob
     end
 
     request = arguments.first[:request]
-    request.update!(response: exception.message, execution_status: SharedConstants::AI_REQUEST_EXECUTION_STATUS[:FAILURE])
+    execution_status_code = SharedConstants::AI_REQUEST_EXECUTION_STATUS[:FAILURE]
+    if exception.message.include? "must have less than 3000 tokens"
```

Comment:
Spent some time looking for a way to parse exception other searching for substring in `message` - printed out `exception.methods` and didn't see anything that would point to this type of error. Open to ideas!

Reply:
Is the type of returned exception any more specific than the standard `Exception`? If so, I think we could add a separate rescue_from (i.e. `rescue_from AwsException do |exception|` if something like that exists) and handle this there.

Reply:
Hey Sanchit - documenting what I attempted even though it unfortunately did not work.

In the server output as it's currently implemented the following is printed:
```
AichatRequestChatCompletionJob Error: /home/ubuntu/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/aws-sdk-core-3.201.3/lib/seahorse/client/plugins/raise_response_errors.rb:17:in `call': Received client error (422) from primary with message "{"error":"Input validation error: `inputs` tokens + `max_new_tokens` must be <= 4096. Given: 3758 `inputs` tokens and 512 `max_new_tokens`","error_type":"validation"}". See https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logEventViewer:group=/aws/sagemaker/Endpoints/gen-ai-mistral-7b-inst-v01 in account 475661607190 for more information. (Aws::SageMakerRuntime::Errors::ModelError)
	from /home
```
Thus, I added the following, similar to the [aws error rescue](https://github.com/code-dot-org/code-dot-org/blob/dff352317734de122d1246763ec5efa7408eb3cc/dashboard/app/jobs/concerns/ai_rubric_config.rb#L123) in `ai_rubric_config`:
```
  rescue Aws::SageMakerRuntime::Errors::ModelError => exception
    if rack_env?(:development)
      puts "AichatRequestChatCompletionJob Error: #{exception.full_message}"
    end

    request = arguments.first[:request]
    execution_status_code = SharedConstants::AI_REQUEST_EXECUTION_STATUS[:FAILURE]
    if exception.message.include? "must have less than 3000 tokens"
      execution_status_code = SharedConstants::AI_REQUEST_EXECUTION_STATUS[:USER_INPUT_TOO_LARGE]
    end

    request.update!(response: exception.message, execution_status: execution_status_code)
    Honeybadger.notify(
      "AichatRequestChatCompletionJob failed with unexpected error: #{exception.message}",
      context: {
        request: request.to_json
      }
    )

    # Raise an exception to notify our system of the failed job.
    raise "AichatRequestChatCompletionJob failed with unexpected error: #{exception.message}. Context: #{request.to_json}"
```
However, I got the following output and `1005` was NOT returned.
` [2024-09-06 18:43:59.152] ERROR -- NotImplementedError: nil`

I tried requiring `aws-sdk-sagemakerruntime` in the file, but then got the following error:
```
NameError at /aichat/start_chat_completion
==========================================
undefined local variable or method `‘aws’ for main:Object
```
I also tried using `rescue_from Aws::SageMakerRuntime::Errors::ModelError do |exception|` and kept the other rescue in. I got status `1000` back. The `rescue` was from the general exception rescue, not the more specific `Aws` error.

I will keep as-is for now, but am definitely open to a follow-up if you have any advice. Thanks!